src/ProgressMeter.jl

```diff
@@ -28,7 +28,7 @@ Holds the five characters that will be used to generate the progress bar.
 mutable struct BarGlyphs
     leftend::Char
     fill::Char
-    front::Char
+    front::Union{Vector{Char}, Char}
```

Comment:
Doesn't it make more sense to have this always be a `Vector{Char}` and just have a
```
BarGlyphs(le::Char, fi::Char, fr::Char, em::Char, re::Char) = BarGlyphs(le, fi, [fr], em, re)
```
constructor for backwards compatibility? That way, anyone using a BarGlyphs can always be sure that `front` is a vector.

Reply:
It is possible but I would have to tweak some things slightly. Right now when the `front` is a vector and we want to draw, say, `2.05` bars, we will notice that for the `0.05` it is better to just print the `empty` than even the first character in the `front` vector.

However, for some bars is not correct (for example the `[=> ]` bar). Here we want to always print the `front` char because it is not a transition between `empty` and `full`.

The change I could make is to require someone to give the `front` vector with the `empty` and `fill` character also included. So it would be `[' ', '▁' ,'▂' ,'▃' ,'▄' ,'▅' ,'▆', '▇', '█']` and make sure that we always print one of the `front`chars. It is a bit redundant perhaps but maybe more general.

Reply:
The reason I think we need the last one is in case we have something like `3.99` bars to show. Then for the default bar we want to show the full character `█` but if we have the restriction that we only show things in `front` then that is not available.

Reply:
Oh, ok, I thought the procedure was to only display the `fill` character once that entire block has been completed (i.e. only display at 4.0). That makes the most sense to me, but it looks like the old way included "round" (in line 419) anyways. I am fine with either rounding or truncating though :)

Reply:
Hm, I don't think we will display all the blocks for the same amount of time in that case.

It seems the current way that is implemented allows one to not have to repeat yourself at the cost of some type checking. In my view it is worth it to provide a simpler interface to the user.

Reply:
Sure, sounds ok to me